accepted/00070-akka.md

```diff
@@ -0,0 +1,134 @@
+- Start Date: 2020-03-12
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Replace homegrown asynchronous task facilities in Java code with an "off-the-shelf", proven open source component (or set of components). Goal is to reduce internal complexity, remove known hacks and possibly attain better observability, performance and ease of tuning.
+
+More background can be found on the [HackWeek 19 Akka project page](https://hackweek.suse.com/projects/uyuni-re-architecting-code-with-akka) which is basically a proof-of-concept of the content of this RFC.
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni's long history has, in some cases, resulted in different organically-grown solutions to problems common in different areas of the codebase. One of those problems is the execution of asynchronous tasks within Java code, where currently we have at least 4 different "frameworks". In addition to the accidental complication, none of them is really complete in terms of current and predictable future requirements.
+
+Specifically, those are:
+
+- Taskomatic's basic facilities, including one-shot tasks
+- Some specific Taskomatic task facilities (eg. `ErrataCacheDriver`, `SSHPushDriver`)
+- Tomcat's `MessageQueue` facility (also replicated in Taskomatic)
+- Tomcat's Salt event handler (`PGEventStream`, which also reuses parts of `MessageQueue`)
+
+![Component diagram](images/00070-components.svg)
+
+Open problems that in various degrees affect all of the above are:
+
+1. enforcing dependencies between asynchronous tasks
+2. enforcing dependencies between asynchronous tasks and other parts of the codebase
+3. establishing priority of tasks
+4. tuning, performance
+5. observability
+6. learnability (by new developers and existing developers unfamiliar with the interested code area, eg., for bugfixing purposes)
+
+At this point the main goal is to **improve internal quality by rationalizing the internal rearchitecture**.
+
+# Detailed design
+[design]: #detailed-design
+
+## Core idea
+
+Remove homegrown solutions and replace them with [akka.io](https://akka.io/), a well-known open source tookit for concurrent/reactive/distributed applications.
+
+With reference to the open problems stated above:
+
+1. in Akka dependencies can be expressed by actors firing messages to other actors when done
+2. any other part of the code base can ask an Akka actor to do something by sending it a message
+3. [Akka has a concept of priority messages to actors](https://doc.akka.io/docs/akka/2.5/mailboxes.html)
+4. Akka seems to be regarded as high performance, has thorough and well-documented tuning knobs
+5. [Akka has a lot of utilities for monitoring](https://developer.lightbend.com/docs/telemetry/current/home.html)
+6. Akka is generally known and well documented, has books, videos and other material available
+
+## Detail design ideas
+
+ - The Akka-based asynchronous facility will be made available via a facade (`ActorManager` in the above PR)
```

Comment:
```suggestion
 - The Akka-based asynchronous facility will be made available via a facade (`ActorManager` in the bellow PR)
```
Link for PR is two paragraphs bellow. Should it be added to the beginning of the test?